### PR TITLE
[runtime] save rpc error code in rpc_tl_query_result_one()

### PR DESCRIPTION
**Pull request by Ramzeeset**
General
----------
Before this pr responses that was gotten from functions `rpc_tl_query_result_one(...)` and `typed_rpc_tl_query_result_one(...)` changed rpc error code to `TL_ERROR_UNKNOWN` independently of it's real error code. It leaded to situations like this
```
array(2) {
  ["__error"]=>
  string(23) "Timeout in KPHP runtime"
  ["__error_code"]=>
  int(-4000)
}
```
But [here](https://github.com/VKCOM/kphp/blob/master/runtime/rpc.cpp#L644) error message composed with a different code

### CODE REVIEW

File: runtime/rpc.cpp [hunk 1]
```diff
@@ -128,6 +134,7 @@ bool f$rpc_parse(const string &new_rpc_data) {
   if (new_rpc_data.size() % sizeof(int) != 0) {
     php_warning("Wrong parameter \"new_rpc_data\" of len %d passed to function rpc_parse", (int)new_rpc_data.size());
     last_rpc_error = "Result's length is not divisible by 4";
+    last_rpc_error_code = TL_ERROR_UNKNOWN;
```
Comment (DrDet) [inline runtime/rpc.cpp:137]:
```suggestion
    last_rpc_error_code = TL_ERROR_RESPONSE_SYNTAX;
```

File: runtime/rpc.cpp [hunk 2]
```diff
@@ -817,16 +825,19 @@ class rpc_get_resumable : public Resumable {
       TRY_WAIT(rpc_get_resumable_label_0, ready, bool);
       if (!ready) {
         last_rpc_error = last_wait_error;
+        last_rpc_error_code = TL_ERROR_UNKNOWN;
```
Comment (DrDet) [inline runtime/rpc.cpp:828]:
```suggestion
        last_rpc_error_code = TL_ERROR_INTERNAL;
```

File: runtime/rpc.cpp [hunk 3]
```diff
@@ -817,16 +825,19 @@ class rpc_get_resumable : public Resumable {
...
         RETURN(false);
       }
 
       Storage *input = get_forked_storage(resumable_id);
       if (input->tag == 0) {
         last_rpc_error = "Result already was gotten";
+        last_rpc_error_code = TL_ERROR_UNKNOWN;
```
Comment (DrDet) [inline runtime/rpc.cpp:835]:
```suggestion
        last_rpc_error_code = TL_ERROR_INTERNAL;
```

File: runtime/rpc.cpp [hunk 4]
```diff
@@ -817,16 +825,19 @@ class rpc_get_resumable : public Resumable {
...
         RETURN(false);
       }
       if (input->tag != Storage::tagger<rpc_request>::get_tag()) {
         last_rpc_error = "Not a rpc request";
+        last_rpc_error_code = TL_ERROR_UNKNOWN;
```
Comment (DrDet) [inline runtime/rpc.cpp:840]:
```suggestion
        last_rpc_error_code = TL_ERROR_INTERNAL;
```

File: runtime/rpc.cpp [hunk 5]
```diff
@@ -890,16 +902,19 @@ class rpc_get_and_parse_resumable : public Resumable {
       TRY_WAIT(rpc_get_and_parse_resumable_label_0, ready, bool);
       if (!ready) {
         last_rpc_error = last_wait_error;
+        last_rpc_error_code = TL_ERROR_UNKNOWN;
```
Comment (DrDet) [inline runtime/rpc.cpp:905]:
```suggestion
        last_rpc_error_code = TL_ERROR_INTERNAL;
```

File: runtime/rpc.cpp [hunk 6]
```diff
@@ -890,16 +902,19 @@ class rpc_get_and_parse_resumable : public Resumable {
...
         RETURN(false);
       }
 
       Storage *input = get_forked_storage(resumable_id);
       if (input->tag == 0) {
         last_rpc_error = "Result already was gotten";
+        last_rpc_error_code = TL_ERROR_UNKNOWN;
```
Comment (DrDet) [inline runtime/rpc.cpp:912]:
```suggestion
        last_rpc_error_code = TL_ERROR_INTERNAL;
```

File: runtime/rpc.cpp [hunk 7]
```diff
@@ -890,16 +902,19 @@ class rpc_get_and_parse_resumable : public Resumable {
...
         RETURN(false);
       }
       if (input->tag != Storage::tagger<rpc_request>::get_tag()) {
         last_rpc_error = "Not a rpc request";
+        last_rpc_error_code = TL_ERROR_UNKNOWN;
```
Comment (DrDet) [inline runtime/rpc.cpp:917]:
```suggestion
        last_rpc_error_code = TL_ERROR_INTERNAL;
```

File: runtime/rpc.cpp [hunk 8]
```diff
@@ -798,6 +805,7 @@ void process_rpc_error(int32_t request_id, int32_t error_code __attribute__((unu
   }
 
   request->error = error_message;
+  request->error_code = error_code;
```
Comment (DrDet) [inline runtime/rpc.cpp:808]: ` __attribute__((unused))` at `error_code` param is not needed anymore

File: runtime/rpc.cpp (hunk 1 above)
Comment (Ramzeeset) [reply runtime/rpc.cpp]: fix

File: runtime/rpc.cpp (hunk 2 above)
Comment (Ramzeeset) [reply runtime/rpc.cpp]: fix

File: runtime/rpc.cpp (hunk 4 above)
Comment (Ramzeeset) [reply runtime/rpc.cpp]: fix

File: runtime/rpc.cpp (hunk 5 above)
Comment (Ramzeeset) [reply runtime/rpc.cpp]: fix

File: runtime/rpc.cpp (hunk 6 above)
Comment (Ramzeeset) [reply runtime/rpc.cpp]: fix

File: runtime/rpc.cpp (hunk 8 above)
Comment (Ramzeeset) [reply runtime/rpc.cpp:808]: fix

File: runtime/rpc.cpp (hunk 7 above)
Comment (Ramzeeset) [reply runtime/rpc.cpp]: fix

**Review by DrDet — APPROVED** [review]

**Pull request merged**